radix-engine-tests/tests/blueprints/pool_protocol_update.rs

```diff
@@ -0,0 +1,1021 @@
+use radix_engine::blueprints::pool::v1::constants::*;
+use radix_engine::blueprints::pool::v1::errors::{
+    multi_resource_pool::Error as MultiResourcePoolError,
+    two_resource_pool::Error as TwoResourcePoolError,
+};
+use radix_engine::system::system_db_reader::*;
+use radix_engine::system::system_type_checker::TypeCheckError;
+use radix_engine_queries::typed_substate_layout::*;
+use scrypto_test::prelude::*;
+use scrypto_unit::*;
+
+#[test]
+fn database_is_consistent_before_and_after_protocol_update() {
```

Comment:
I've added a folder specifically for protocol_update / flash tests. Can we add this here: `radix-engine-tests/tets/flash/`